src/hausdorff.jl

```diff
@@ -123,11 +133,10 @@ function pairwise(d::GenericHausdorff,
 
     for j=1:n
       psetB = psetsB[j]
-      for i=1:min(j-1, m)
+      for i=1:min(j, m)
         psetA = psetsA[i]
         D[i,j] = evaluate_pset(d, psetA, psetB)
       end
-      i==m && continue
```

Comment:
I see that you are evaluating the calculation along the diagonal. Isn't that a waste? If we have many images, the calculation can be expensive.

Reply:
This function is introduced in https://github.com/JuliaImages/ImageDistances.jl/commit/7a54c14a4a67b96b9c17698aa181dff4d1de3fc0, but it would raise NotInBound error if `m ~= n`. Looks like the patch I made in https://github.com/JuliaImages/ImageDistances.jl/commit/ccde15b87b37b488d1f759765980c3b3dc3e82a9 is a regression.

I'll check the details when I'm available.